gtda/externals/python/ripser_interface.py

```diff
@@ -158,8 +160,56 @@ def _resolve_symmetry_conflicts(coo):
         return _row, _col, _data
 
 
+def _compute_dtm_weights(dm, n_neighbors, weights_r):
+    knn = kneighbors_graph(dm, n_neighbors=n_neighbors,
+                           metric="precomputed", mode="distance",
+                           include_self=False)
+
+    return 2 * (
+            np.asarray(knn.power(weights_r)).sum(axis=1) / n_neighbors
+            ) ** (1 / weights_r)
+
+
+def _weigh_filtration(distances, weights_x, weights_y, p):
+    """Create a DTM-weighted distance matrix. For dense data, `weights_x` is
+    a column vector, `weights_y` is a 1D array, `distances` is the original
+    original distance matrix, and the computations exploit array broadcasting.
+    For sparse data, all three are 1D arrays. `p` can only be ``numpy.inf``,
+    ``1``, or ``2``."""
+    if p == np.inf:
+        return np.maximum(distances, np.maximum(weights_x, weights_y))
+    elif p == 1:
+        return np.where(distances <= np.abs(weights_x - weights_y) / 2,
+                        np.maximum(weights_x, weights_y),
+                        distances + (weights_x + weights_y) / 2)
+    elif p == 2:
+        return np.where(
+            distances <= np.abs(weights_x**2 - weights_y**2)**.5 / 2,
+            np.maximum(weights_x, weights_y),
+            np.sqrt((distances**2 + ((weights_x + weights_y) / 2)**2) *
+                    (distances**2 + ((weights_x - weights_y) / 2)**2)) /
+            distances
+            )
```

Comment:
Looking at proposition 3.5, i see the `0.5` here:
 ```
       return np.where(
            distances <= np.abs(weights_x**2 - weights_y**2)**.5 / 2,
            np.maximum(weights_x, weights_y),
            np.sqrt((distances**2 + ((weights_x + weights_y) )**2) *
                    (distances**2 + ((weights_x - weights_y) )**2)) /
           (2* distances)
            )

```

Reply:
There are a lot of factor 1/2 differences between the paper and this implementation. They come from the fact that the paper uses a "Cech"-type convention (even for Rips), while I think we should be in line with the Vietoris-Rips convention. In particular, passing zero weights (zero weight function) should yield vanilla VR.

Reply:
Basically, the current code should yield PDs where are births and all deaths are twice the births and deaths from the original paper. It is also important that the formulae here also are the appropriate ones when one passes custom weights.

Reply:
Ok, I will double check, then! Sorry for the false alarm !